common/text_window.py

```diff
@@ -31,7 +32,8 @@ def close(self):
 
   def wait_for_exit(self):
     if self.text_proc is not None:
-      while True:
+      params = Params()
+      while not params.get_bool('DoShutdown'):
```

Comment:
This could halt during an exception

Reply:
Unsure about this change. Since the CI seemed to crash on the import line.
I moved it to the function and CI is passing, but I think this is just because it isn't executed in any test